App crashed when I opened  the data base and tried to upload a non MMB file

***App version***: Version. 2019.04.10.1000-beta(1000)
### Description

App crashed when I opened data base and wanted to upload an image.

### Expected behavior

I should be able to upload image when I opened the data base and clicked on other.

### Steps to reproduce

- Launch the app
- Click on the menu button BG the top left side
- Click on data base
- Click on other and select an image. The app will crash.

### Reproducing the behaviour

https://youtu.be/a1txwhA-BIM

Device used: Android 7.0 Techno spark K7

``` FATAL EXCEPTION: main
05-08 09:15:48.412  9117  9117 E AndroidRuntime: Process: com.money.manager.ex, PID: 9117
05-08 09:15:48.412  9117  9117 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=15, result=-1, data=Intent { dat=content://com.android.providers.media.documents/document/image:5309 flg=0x43 }} to activity {com.money.manager.ex/com.money.manager.ex.home.MainActivity}: java.lang.IllegalArgumentException: Not a valid database file!
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4178)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4221)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread.-wrap20(ActivityThread.java)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1583)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:110)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:203)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6251)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1063)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Not a valid database file!
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.money.manager.ex.utils.MmxDatabaseUtils.useDatabase(MmxDatabaseUtils.java:207)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.money.manager.ex.core.docstorage.FileStorageHelper.pullDatabase(FileStorageHelper.java:207)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.money.manager.ex.core.docstorage.FileStorageHelper.selectDatabase(FileStorageHelper.java:92)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at com.money.manager.ex.home.MainActivity.onActivityResult(MainActivity.java:318)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:6928)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4174)
05-08 09:15:48.412  9117  9117 E AndroidRuntime: 	... 9 more```

--- a/app/src/main/java/com/money/manager/ex/core/docstorage/FileStorageHelper.java
+++ b/app/src/main/java/com/money/manager/ex/core/docstorage/FileStorageHelper.java
@@ -11,6 +11,7 @@ import android.os.Looper;
 import android.os.ParcelFileDescriptor;
 import android.provider.DocumentsContract;
 import android.provider.OpenableColumns;
+import android.widget.Toast;
 
 import androidx.appcompat.app.AppCompatActivity;
 
@@ -199,7 +200,15 @@ public class FileStorageHelper {
 
         // store the metadata.
         MmxDatabaseUtils dbUtils = new MmxDatabaseUtils(getContext());
-        dbUtils.useDatabase(metadata);
+
+        // issue #1359
+        try {
+            dbUtils.useDatabase(metadata);
+        } catch (Exception e) {
+            Timber.e(e);
+            Toast.makeText(getContext(),"Unable to open DB. Not a .mmb file.", Toast.LENGTH_SHORT).show();
+            return;
+        }
     }
 
     /**

--- a/app/src/main/java/com/money/manager/ex/home/MainActivity.java
+++ b/app/src/main/java/com/money/manager/ex/home/MainActivity.java
@@ -319,7 +319,9 @@ public class MainActivity
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
 
-        if (resultCode != RESULT_OK) return;
+        // don't accidentally bypass passcode (failures), e.g. pressing physical back button
+        // see old merge #1338 and issue #1293
+        if (resultCode != RESULT_OK && requestCode != RequestCodes.PASSCODE) return;
 
         switch (requestCode) {
             case RequestCodes.SELECT_DOCUMENT:

